### DISCUSSION TITLE
I'm wondering if it's possible to do a bulk search by collecting queries from a certain interval instead of responding to search queries one by one.

### DISCUSSION THREAD

**Discussion by wookiist** [milvus-io/milvus#30762] [Q&A and General discussion] [answered]
I'm wondering if there is a way to do a bulk search in milvus when there is a sudden influx of search queries, instead of processing them one by one immediately, but with a short interval of delay, and then collecting the search queries that came in during that interval and doing a bulk search. 

Is it possible with milvus?

**Comment by yhmo** [comment]
Milvus internally has "search combine" machinery to execute a batch of search requests in one task.
Assume you have 100 clients, sending 100 search requests at the same time to the Milvus server. They could be combined as a single task to execute, and the result contains 100 * topk items, then the 100 * topk items are returned to 100 clients respectively.

**Reply by wookiist** [reply]
Thank you. 
Is it possible to know how many time intervals correspond to the "at the same time" you mentioned?
For example, if there is a strategy to combine queries that come in within 10ms together.

**Reply by yhmo** [reply]
No const intervals.
It depends on runtime behavior. If the previous search request is still in process, the next received requests will be accumulated in a queue. If these requests have similar parameters, they will be merged into a large request to execute.

**Comment by xiaofan-luan** [comment, marked as answer]
This is called dynamic batching.

The batching happened when you many small nq request in queue and then have same filter expressions

**Reply by wookiist** [reply]
Sounds great, can you share some code or a document with more explanation to see what the criteria is internally? Thanks.
